fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBQueriedRecord.java

```diff
@@ -62,12 +65,68 @@
     @Nullable
     public abstract IndexEntry getIndexEntry();
 
+    /**
+     * Get the synthetic record if this is a synthetic record type..
+     * @return a synthetic record of {@code null}
+     */
+    @Nullable
+    public abstract FDBSyntheticRecord getSyntheticRecord();
+
+    /**
+     * Get the name of this constituent if it comes from a synthetic record.
+     * @return a constituent name of {@code null}
+     */
+    @Nullable
+    public abstract String getConstituentName();
+
     /**
      * The constituents of this record if it is a synthetic record type.
      * @return a map of constituents from name to {@link FDBStoredRecord}
      */
     @Nullable
-    public abstract Map<String, FDBStoredRecord<? extends Message>> getConstituents();
+    public Map<String, FDBStoredRecord<? extends Message>> getConstituents() {
```

Comment:
Can we put this into `SyntheticRecord`? Seems like a bunch of stuff logic that nulls out if `getSyntheticRecord()` nulls out.